source/libip6string/ip6tos.c

```diff
@@ -92,4 +95,27 @@ void ip6tos(const void *ip6addr, char *p)
         }
     }
     *p++ = '\0';
+
+    // Return length of generated string, excluding the terminating null-character
+    return p - p_orig - 1;
```

Comment:
Could not increment p in the first place. Not that it would affect the compiled code, I guess.